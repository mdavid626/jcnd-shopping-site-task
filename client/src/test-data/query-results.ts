export const productsQueryResult = {
  data: {
    products: {
      nodes: [
        {
          id: '63ea6485732ff5cf8b364431',
          priceInCents: 9699,
          name: 'Broccoli',
          inStock: 466,
          __typename: 'Product',
        },
        {
          id: '63ea6485732ff5cf8b364432',
          priceInCents: 3434,
          name: 'Broccolini',
          inStock: 223,
          __typename: 'Product',
        },
        {
          id: '63ea6485732ff5cf8b364433',
          priceInCents: 3692,
          name: 'Brussels Sprouts',
          inStock: 12,
          __typename: 'Product',
        },
        {
          id: '63ea6485732ff5cf8b364434',
          priceInCents: 8839,
          name: 'Cabbage',
          inStock: 200,
          __typename: 'Product',
        },
        {
          id: '63ea6485732ff5cf8b364435',
          priceInCents: 6828,
          name: 'Carrots',
          inStock: 151,
          __typename: 'Product',
        },
      ],
      pageInfo: { page: 1, numberOfPages: 15, __typename: 'PageInfo' },
      __typename: 'ProductsQueryResult',
    },
  },
};

export const placeOrderQueryResult1 = {
  data: { placeOrder: '63ede5e031f3fc4012a0951b' },
};
