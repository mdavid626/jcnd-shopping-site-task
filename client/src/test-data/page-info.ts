import { PageInfo } from '../types/page-info';

export const testPageInfo1: PageInfo = {
  page: 1,
  numberOfPages: 5,
};
