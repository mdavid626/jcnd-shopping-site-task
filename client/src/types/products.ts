import { PageInfo } from './page-info';

export type ProductCategory = 'VEGETABLES' | 'FRUITS' | 'CHEESE';

export type Product = {
  id: string;
  price: number;
  name: string;
  inStock: number;
};

export type ProductsQueryResult = {
  products: { nodes: Product[]; pageInfo: PageInfo };
};
