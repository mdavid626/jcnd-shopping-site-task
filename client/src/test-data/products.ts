import { Product, ProductsQueryResult } from '../types/products';

export const testProduct1: Product = {
  id: '63ea6485732ff5cf8b36442c',
  name: 'Artichoke',
  price: 46.36,
  inStock: 5,
};

export const testProduct2: Product = {
  id: '63ea6485732ff5cf8b36442d',
  name: 'Asparagus',
  price: 25.36,
  inStock: 3,
};

export const testProductQueryResult1: ProductsQueryResult = {
  products: {
    nodes: [testProduct1, testProduct2],
    pageInfo: {
      page: 1,
      numberOfPages: 2,
    },
  },
};
