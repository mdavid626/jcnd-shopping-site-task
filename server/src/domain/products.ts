import { ProductCategory } from '../types/product';
import sampleProducts from './sample-products';

const getProducts = (category: ProductCategory) => {
  return sampleProducts.filter((product) => product.category === category);
};

export default {
  getProducts,
};
