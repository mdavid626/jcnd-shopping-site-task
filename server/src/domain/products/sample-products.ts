import { Product } from '../../types/product';

const vegetables: Product[] = [
  {
    id: '63ea6485732ff5cf8b36442c',
    name: 'Artichoke',
    category: 'VEGETABLES',
    price: 46.36,
    inStock: 5,
  },
  {
    id: '63ea6485732ff5cf8b36442d',
    name: 'Asparagus',
    category: 'VEGETABLES',
    price: 25.36,
    inStock: 3,
  },
  {
    id: '63ea6485732ff5cf8b36442e',
    name: 'Eggplant',
    category: 'VEGETABLES',
    price: 46.31,
    inStock: 25,
  },
  {
    id: '63ea6485732ff5cf8b36442f',
    name: 'Avocado',
    category: 'VEGETABLES',
    price: 38.38,
    inStock: 977,
  },
  {
    id: '63ea6485732ff5cf8b364430',
    name: 'Beet',
    category: 'VEGETABLES',
    price: 43.61,
    inStock: 985,
  },
  {
    id: '63ea6485732ff5cf8b364431',
    name: 'Broccoli',
    category: 'VEGETABLES',
    price: 96.99,
    inStock: 466,
  },
  {
    id: '63ea6485732ff5cf8b364432',
    name: 'Broccolini',
    category: 'VEGETABLES',
    price: 34.34,
    inStock: 223,
  },
  {
    id: '63ea6485732ff5cf8b364433',
    name: 'Brussels Sprouts',
    category: 'VEGETABLES',
    price: 36.92,
    inStock: 12,
  },
  {
    id: '63ea6485732ff5cf8b364434',
    name: 'Cabbage',
    category: 'VEGETABLES',
    price: 88.39,
    inStock: 200,
  },
  {
    id: '63ea6485732ff5cf8b364435',
    name: 'Carrots',
    category: 'VEGETABLES',
    price: 68.28,
    inStock: 151,
  },
  {
    id: '63ea6485732ff5cf8b364436',
    name: 'Cauliflower',
    category: 'VEGETABLES',
    price: 51.71,
    inStock: 119,
  },
  {
    id: '63ea6485732ff5cf8b364437',
    name: 'Celeriac',
    category: 'VEGETABLES',
    price: 49.25,
    inStock: 16,
  },
  {
    id: '63ea6485732ff5cf8b364438',
    name: 'Celery',
    category: 'VEGETABLES',
    price: 48.14,
    inStock: 713,
  },
  {
    id: '63ea6485732ff5cf8b364439',
    name: 'Chicory',
    category: 'VEGETABLES',
    price: 5.56,
    inStock: 494,
  },
  {
    id: '63ea6485732ff5cf8b36443a',
    name: 'Corn',
    category: 'VEGETABLES',
    price: 99.31,
    inStock: 351,
  },
  {
    id: '63ea6485732ff5cf8b36443b',
    name: 'Cress',
    category: 'VEGETABLES',
    price: 52.19,
    inStock: 703,
  },
  {
    id: '63ea6485732ff5cf8b36443c',
    name: 'Cucumbers',
    category: 'VEGETABLES',
    price: 84.36,
    inStock: 689,
  },
  {
    id: '63ea6485732ff5cf8b36443d',
    name: 'Garlic',
    category: 'VEGETABLES',
    price: 54.35,
    inStock: 202,
  },
  {
    id: '63ea6485732ff5cf8b36443e',
    name: 'Greens',
    category: 'VEGETABLES',
    price: 38.62,
    inStock: 208,
  },
  {
    id: '63ea6485732ff5cf8b36443f',
    name: 'Gourds',
    category: 'VEGETABLES',
    price: 7.53,
    inStock: 330,
  },
  {
    id: '63ea6485732ff5cf8b364440',
    name: 'Jicama',
    category: 'VEGETABLES',
    price: 17.7,
    inStock: 497,
  },
  {
    id: '63ea6485732ff5cf8b364441',
    name: 'Kale',
    category: 'VEGETABLES',
    price: 27.52,
    inStock: 931,
  },
  {
    id: '63ea6485732ff5cf8b364442',
    name: 'Kohlrabi',
    category: 'VEGETABLES',
    price: 22.84,
    inStock: 158,
  },
  {
    id: '63ea6485732ff5cf8b364443',
    name: 'Leeks',
    category: 'VEGETABLES',
    price: 75.1,
    inStock: 50,
  },
  {
    id: '63ea6485732ff5cf8b364444',
    name: 'Mushrooms',
    category: 'VEGETABLES',
    price: 63.55,
    inStock: 867,
  },
  {
    id: '63ea6485732ff5cf8b364445',
    name: 'Okra',
    category: 'VEGETABLES',
    price: 73.5,
    inStock: 439,
  },
  {
    id: '63ea6485732ff5cf8b364446',
    name: 'Onions',
    category: 'VEGETABLES',
    price: 2.4,
    inStock: 297,
  },
  {
    id: '63ea6485732ff5cf8b364447',
    name: 'Parsnips',
    category: 'VEGETABLES',
    price: 63.17,
    inStock: 266,
  },
  {
    id: '63ea6485732ff5cf8b364448',
    name: 'Peas',
    category: 'VEGETABLES',
    price: 35.64,
    inStock: 207,
  },
  {
    id: '63ea6485732ff5cf8b364449',
    name: 'Potatoes',
    category: 'VEGETABLES',
    price: 28.26,
    inStock: 479,
  },
  {
    id: '63ea6485732ff5cf8b36444a',
    name: 'Pumpkin',
    category: 'VEGETABLES',
    price: 28.12,
    inStock: 676,
  },
  {
    id: '63ea6485732ff5cf8b36444b',
    name: 'Radicchio',
    category: 'VEGETABLES',
    price: 3.95,
    inStock: 437,
  },
  {
    id: '63ea6485732ff5cf8b36444c',
    name: 'Radish',
    category: 'VEGETABLES',
    price: 44.3,
    inStock: 905,
  },
  {
    id: '63ea6485732ff5cf8b36444d',
    name: 'Rhubarb',
    category: 'VEGETABLES',
    price: 47.11,
    inStock: 639,
  },
  {
    id: '63ea6485732ff5cf8b36444e',
    name: 'Swede',
    category: 'VEGETABLES',
    price: 12.46,
    inStock: 275,
  },
  {
    id: '63ea6485732ff5cf8b36444f',
    name: 'Turnips',
    category: 'VEGETABLES',
    price: 83.02,
    inStock: 551,
  },
  {
    id: '63ea6485732ff5cf8b364450',
    name: 'Artichoke',
    category: 'VEGETABLES',
    price: 75.32,
    inStock: 64,
  },
  {
    id: '63ea6485732ff5cf8b364451',
    name: 'Asparagus',
    category: 'VEGETABLES',
    price: 73.96,
    inStock: 249,
  },
  {
    id: '63ea6485732ff5cf8b364452',
    name: 'Eggplant',
    category: 'VEGETABLES',
    price: 72.78,
    inStock: 815,
  },
  {
    id: '63ea6485732ff5cf8b364453',
    name: 'Avocado',
    category: 'VEGETABLES',
    price: 38.86,
    inStock: 386,
  },
  {
    id: '63ea6485732ff5cf8b364454',
    name: 'Beet',
    category: 'VEGETABLES',
    price: 68.36,
    inStock: 887,
  },
  {
    id: '63ea6485732ff5cf8b364455',
    name: 'Broccoli',
    category: 'VEGETABLES',
    price: 40.35,
    inStock: 793,
  },
  {
    id: '63ea6485732ff5cf8b364456',
    name: 'Broccolini',
    category: 'VEGETABLES',
    price: 50.23,
    inStock: 130,
  },
  {
    id: '63ea6485732ff5cf8b364457',
    name: 'Brussels Sprouts',
    category: 'VEGETABLES',
    price: 19.09,
    inStock: 462,
  },
  {
    id: '63ea6485732ff5cf8b364458',
    name: 'Cabbage',
    category: 'VEGETABLES',
    price: 24.92,
    inStock: 258,
  },
  {
    id: '63ea6485732ff5cf8b364459',
    name: 'Carrots',
    category: 'VEGETABLES',
    price: 17.48,
    inStock: 515,
  },
  {
    id: '63ea6485732ff5cf8b36445a',
    name: 'Cauliflower',
    category: 'VEGETABLES',
    price: 58.73,
    inStock: 601,
  },
  {
    id: '63ea6485732ff5cf8b36445b',
    name: 'Celeriac',
    category: 'VEGETABLES',
    price: 81.04,
    inStock: 138,
  },
  {
    id: '63ea6485732ff5cf8b36445c',
    name: 'Celery',
    category: 'VEGETABLES',
    price: 76.45,
    inStock: 857,
  },
  {
    id: '63ea6485732ff5cf8b36445d',
    name: 'Chicory',
    category: 'VEGETABLES',
    price: 94.49,
    inStock: 217,
  },
  {
    id: '63ea6485732ff5cf8b36445e',
    name: 'Corn',
    category: 'VEGETABLES',
    price: 37.41,
    inStock: 787,
  },
  {
    id: '63ea6485732ff5cf8b36445f',
    name: 'Cress',
    category: 'VEGETABLES',
    price: 63.46,
    inStock: 962,
  },
  {
    id: '63ea6485732ff5cf8b364460',
    name: 'Cucumbers',
    category: 'VEGETABLES',
    price: 69.03,
    inStock: 414,
  },
  {
    id: '63ea6485732ff5cf8b364461',
    name: 'Garlic',
    category: 'VEGETABLES',
    price: 82.71,
    inStock: 947,
  },
  {
    id: '63ea6485732ff5cf8b364462',
    name: 'Greens',
    category: 'VEGETABLES',
    price: 17.61,
    inStock: 536,
  },
  {
    id: '63ea6485732ff5cf8b364463',
    name: 'Gourds',
    category: 'VEGETABLES',
    price: 15.68,
    inStock: 950,
  },
  {
    id: '63ea6485732ff5cf8b364464',
    name: 'Jicama',
    category: 'VEGETABLES',
    price: 58.61,
    inStock: 458,
  },
  {
    id: '63ea6485732ff5cf8b364465',
    name: 'Kale',
    category: 'VEGETABLES',
    price: 61.48,
    inStock: 485,
  },
  {
    id: '63ea6485732ff5cf8b364466',
    name: 'Kohlrabi',
    category: 'VEGETABLES',
    price: 39.38,
    inStock: 166,
  },
  {
    id: '63ea6485732ff5cf8b364467',
    name: 'Leeks',
    category: 'VEGETABLES',
    price: 98.45,
    inStock: 385,
  },
  {
    id: '63ea6485732ff5cf8b364468',
    name: 'Mushrooms',
    category: 'VEGETABLES',
    price: 9.86,
    inStock: 862,
  },
  {
    id: '63ea6485732ff5cf8b364469',
    name: 'Okra',
    category: 'VEGETABLES',
    price: 28.7,
    inStock: 396,
  },
  {
    id: '63ea6485732ff5cf8b36446a',
    name: 'Onions',
    category: 'VEGETABLES',
    price: 63.66,
    inStock: 357,
  },
  {
    id: '63ea6485732ff5cf8b36446b',
    name: 'Parsnips',
    category: 'VEGETABLES',
    price: 94.53,
    inStock: 318,
  },
  {
    id: '63ea6485732ff5cf8b36446c',
    name: 'Peas',
    category: 'VEGETABLES',
    price: 43.2,
    inStock: 407,
  },
  {
    id: '63ea6485732ff5cf8b36446d',
    name: 'Potatoes',
    category: 'VEGETABLES',
    price: 86.57,
    inStock: 215,
  },
  {
    id: '63ea6485732ff5cf8b36446e',
    name: 'Pumpkin',
    category: 'VEGETABLES',
    price: 12.26,
    inStock: 530,
  },
  {
    id: '63ea6485732ff5cf8b36446f',
    name: 'Radicchio',
    category: 'VEGETABLES',
    price: 58.57,
    inStock: 80,
  },
  {
    id: '63ea6485732ff5cf8b364470',
    name: 'Radish',
    category: 'VEGETABLES',
    price: 92.43,
    inStock: 763,
  },
  {
    id: '63ea6485732ff5cf8b364471',
    name: 'Rhubarb',
    category: 'VEGETABLES',
    price: 8.43,
    inStock: 809,
  },
  {
    id: '63ea6485732ff5cf8b364472',
    name: 'Swede',
    category: 'VEGETABLES',
    price: 80.13,
    inStock: 988,
  },
  {
    id: '63ea6485732ff5cf8b364473',
    name: 'Turnips',
    category: 'VEGETABLES',
    price: 11.3,
    inStock: 7,
  },
];

const fruits: Product[] = [
  {
    id: '63ea65b5e0bff71580e39a68',
    name: 'Apple',
    category: 'FRUITS',
    price: 46.49,
    inStock: 505,
  },
  {
    id: '63ea65b5e0bff71580e39a69',
    name: 'Banana',
    category: 'FRUITS',
    price: 41.25,
    inStock: 867,
  },
  {
    id: '63ea65b5e0bff71580e39a6a',
    name: 'Orange',
    category: 'FRUITS',
    price: 34.9,
    inStock: 610,
  },
  {
    id: '63ea65b5e0bff71580e39a6b',
    name: 'Lemon',
    category: 'FRUITS',
    price: 17.45,
    inStock: 814,
  },
  {
    id: '63ea65b5e0bff71580e39a6c',
    name: 'Lime',
    category: 'FRUITS',
    price: 48.92,
    inStock: 359,
  },
  {
    id: '63ea65b5e0bff71580e39a6d',
    name: 'Melon',
    category: 'FRUITS',
    price: 46.84,
    inStock: 496,
  },
  {
    id: '63ea65b5e0bff71580e39a6e',
    name: 'Watermelon',
    category: 'FRUITS',
    price: 90.53,
    inStock: 200,
  },
  {
    id: '63ea65b5e0bff71580e39a6f',
    name: 'Grape',
    category: 'FRUITS',
    price: 59.23,
    inStock: 42,
  },
  {
    id: '63ea65b5e0bff71580e39a70',
    name: 'Grapefruit',
    category: 'FRUITS',
    price: 89.18,
    inStock: 836,
  },
  {
    id: '63ea65b5e0bff71580e39a71',
    name: 'Plum',
    category: 'FRUITS',
    price: 25.73,
    inStock: 992,
  },
  {
    id: '63ea65b5e0bff71580e39a72',
    name: 'Mango',
    category: 'FRUITS',
    price: 76.9,
    inStock: 667,
  },
  {
    id: '63ea65b5e0bff71580e39a73',
    name: 'Pineapple',
    category: 'FRUITS',
    price: 64.03,
    inStock: 681,
  },
  {
    id: '63ea65b5e0bff71580e39a74',
    name: 'Kiwi',
    category: 'FRUITS',
    price: 87.17,
    inStock: 857,
  },
  {
    id: '63ea65b5e0bff71580e39a75',
    name: 'Peach',
    category: 'FRUITS',
    price: 47.39,
    inStock: 462,
  },
  {
    id: '63ea65b5e0bff71580e39a76',
    name: 'Avocado',
    category: 'FRUITS',
    price: 39.54,
    inStock: 771,
  },
  {
    id: '63ea65b5e0bff71580e39a77',
    name: 'Coconut',
    category: 'FRUITS',
    price: 4.48,
    inStock: 905,
  },
  {
    id: '63ea65b5e0bff71580e39a78',
    name: 'Apricot',
    category: 'FRUITS',
    price: 59.75,
    inStock: 986,
  },
  {
    id: '63ea65b5e0bff71580e39a79',
    name: 'Blueberry',
    category: 'FRUITS',
    price: 99.72,
    inStock: 45,
  },
  {
    id: '63ea65b5e0bff71580e39a7a',
    name: 'Blackberry',
    category: 'FRUITS',
    price: 52.7,
    inStock: 482,
  },
  {
    id: '63ea65b5e0bff71580e39a7b',
    name: 'Strawberry',
    category: 'FRUITS',
    price: 51.65,
    inStock: 684,
  },
  {
    id: '63ea65b5e0bff71580e39a7c',
    name: 'Raspberry',
    category: 'FRUITS',
    price: 98.67,
    inStock: 145,
  },
  {
    id: '63ea65b5e0bff71580e39a7d',
    name: 'Cherry',
    category: 'FRUITS',
    price: 92.16,
    inStock: 194,
  },
  {
    id: '63ea65b5e0bff71580e39a7e',
    name: 'Pear',
    category: 'FRUITS',
    price: 39.65,
    inStock: 352,
  },
  {
    id: '63ea65b5e0bff71580e39a7f',
    name: 'Mandarin',
    category: 'FRUITS',
    price: 69.23,
    inStock: 34,
  },
  {
    id: '63ea65b5e0bff71580e39a80',
    name: 'Durian',
    category: 'FRUITS',
    price: 82.04,
    inStock: 565,
  },
  {
    id: '63ea65b5e0bff71580e39a81',
    name: 'Passion fruit',
    category: 'FRUITS',
    price: 33.24,
    inStock: 734,
  },
  {
    id: '63ea65b5e0bff71580e39a82',
    name: 'Persimmon',
    category: 'FRUITS',
    price: 86.97,
    inStock: 833,
  },
  {
    id: '63ea65b5e0bff71580e39a83',
    name: 'Dragon fruit',
    category: 'FRUITS',
    price: 85.96,
    inStock: 418,
  },
  {
    id: '63ea65b5e0bff71580e39a84',
    name: 'Papaya',
    category: 'FRUITS',
    price: 87.29,
    inStock: 715,
  },
  {
    id: '63ea65b5e0bff71580e39a85',
    name: 'Pomegranate',
    category: 'FRUITS',
    price: 44.53,
    inStock: 620,
  },
];

const cheese: Product[] = [
  {
    id: '63ea6652e0bff71580e39a86',
    name: 'Affineur',
    category: 'CHEESE',
    price: 24.9,
    inStock: 52,
  },
  {
    id: '63ea6652e0bff71580e39a87',
    name: 'Bergkäse',
    category: 'CHEESE',
    price: 46.76,
    inStock: 574,
  },
  {
    id: '63ea6652e0bff71580e39a88',
    name: 'Brimsen',
    category: 'CHEESE',
    price: 48.41,
    inStock: 824,
  },
  {
    id: '63ea6652e0bff71580e39a89',
    name: 'Dachsteiner',
    category: 'CHEESE',
    price: 17.52,
    inStock: 462,
  },
  {
    id: '63ea6652e0bff71580e39a8a',
    name: 'Lüneberg cheese',
    category: 'CHEESE',
    price: 35.47,
    inStock: 289,
  },
  {
    id: '63ea6652e0bff71580e39a8b',
    name: 'Tyrolean grey cheese',
    category: 'CHEESE',
    price: 30.01,
    inStock: 484,
  },
  {
    id: '63ea6652e0bff71580e39a8c',
    name: 'Beauvoorde cheese',
    category: 'CHEESE',
    price: 77.08,
    inStock: 832,
  },
  {
    id: '63ea6652e0bff71580e39a8d',
    name: 'Herve cheese',
    category: 'CHEESE',
    price: 29.82,
    inStock: 896,
  },
  {
    id: '63ea6652e0bff71580e39a8e',
    name: 'Limburger cheese',
    category: 'CHEESE',
    price: 7.84,
    inStock: 8,
  },
  {
    id: '63ea6652e0bff71580e39a8f',
    name: 'Maredsous cheese',
    category: 'CHEESE',
    price: 82.8,
    inStock: 980,
  },
  {
    id: '63ea6652e0bff71580e39a90',
    name: 'Passendale cheese',
    category: 'CHEESE',
    price: 67.59,
    inStock: 159,
  },
  {
    id: '63ea6652e0bff71580e39a91',
    name: 'Postel cheese',
    category: 'CHEESE',
    price: 68.54,
    inStock: 936,
  },
  {
    id: '63ea6652e0bff71580e39a92',
    name: 'Prince-Jean cheese',
    category: 'CHEESE',
    price: 40.93,
    inStock: 791,
  },
  {
    id: '63ea6652e0bff71580e39a93',
    name: 'Remedou cheese',
    category: 'CHEESE',
    price: 85.24,
    inStock: 119,
  },
  {
    id: '63ea6652e0bff71580e39a94',
    name: 'Rodoric cheese',
    category: 'CHEESE',
    price: 32.52,
    inStock: 457,
  },
  {
    id: '63ea6652e0bff71580e39a95',
    name: 'Rubens cheese',
    category: 'CHEESE',
    price: 83.96,
    inStock: 612,
  },
  {
    id: '63ea6652e0bff71580e39a96',
    name: 'Aura',
    category: 'CHEESE',
    price: 76.68,
    inStock: 99,
  },
  {
    id: '63ea6652e0bff71580e39a97',
    name: 'Lappi cheese',
    category: 'CHEESE',
    price: 70.05,
    inStock: 234,
  },
  {
    id: '63ea6652e0bff71580e39a98',
    name: 'Leipäjuusto (Cheese bread)',
    category: 'CHEESE',
    price: 27.18,
    inStock: 616,
  },
  {
    id: '63ea6652e0bff71580e39a99',
    name: 'Raejuusto (Finnish cottage cheese)',
    category: 'CHEESE',
    price: 42.93,
    inStock: 45,
  },
  {
    id: '63ea6652e0bff71580e39a9a',
    name: 'Uunijuusto (Baked cheese)',
    category: 'CHEESE',
    price: 23.8,
    inStock: 447,
  },
  {
    id: '63ea6652e0bff71580e39a9b',
    name: 'Bavaria Blu',
    category: 'CHEESE',
    price: 96.54,
    inStock: 499,
  },
  {
    id: '63ea6652e0bff71580e39a9c',
    name: 'Bördespeck',
    category: 'CHEESE',
    price: 73.4,
    inStock: 969,
  },
  {
    id: '63ea6652e0bff71580e39a9d',
    name: 'Cambozola',
    category: 'CHEESE',
    price: 50.67,
    inStock: 329,
  },
  {
    id: '63ea6652e0bff71580e39a9e',
    name: 'Emmental',
    category: 'CHEESE',
    price: 41.32,
    inStock: 64,
  },
  {
    id: '63ea6652e0bff71580e39a9f',
    name: 'Grünländer',
    category: 'CHEESE',
    price: 35.64,
    inStock: 919,
  },
  {
    id: '63ea6652e0bff71580e39aa0',
    name: 'Harzer cheese',
    category: 'CHEESE',
    price: 56.82,
    inStock: 967,
  },
  {
    id: '63ea6652e0bff71580e39aa1',
    name: 'Handkäse',
    category: 'CHEESE',
    price: 89.38,
    inStock: 491,
  },
  {
    id: '63ea6652e0bff71580e39aa2',
    name: 'Klützer Gold',
    category: 'CHEESE',
    price: 53.92,
    inStock: 167,
  },
  {
    id: '63ea6652e0bff71580e39aa3',
    name: 'Limburger cheese',
    category: 'CHEESE',
    price: 79.89,
    inStock: 80,
  },
  {
    id: '63ea6652e0bff71580e39aa4',
    name: 'Romadur',
    category: 'CHEESE',
    price: 2.58,
    inStock: 904,
  },
  {
    id: '63ea6652e0bff71580e39aa5',
    name: 'Tilsit cheese',
    category: 'CHEESE',
    price: 13.68,
    inStock: 485,
  },
  {
    id: '63ea6652e0bff71580e39aa6',
    name: 'Spundekäs cheese',
    category: 'CHEESE',
    price: 48.22,
    inStock: 172,
  },
  {
    id: '63ea6652e0bff71580e39aa7',
    name: 'Steinbuscher',
    category: 'CHEESE',
    price: 15.62,
    inStock: 111,
  },
  {
    id: '63ea6652e0bff71580e39aa8',
    name: 'Weisslacker',
    category: 'CHEESE',
    price: 78.36,
    inStock: 41,
  },
];

const products: Product[] = [...vegetables, ...fruits, ...cheese];

export default products;
