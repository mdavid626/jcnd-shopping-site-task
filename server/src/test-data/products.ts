import { Product, ProductsQueryResult } from '../types/product';

const testProduct1: Product = {
  id: '63ea6485732ff5cf8b36442c',
  name: 'Artichoke',
  category: 'VEGETABLES',
  priceInCents: 123,
  inStock: 5,
};

const testProduct2: Product = {
  id: '63ea6485732ff5cf8b36442d',
  name: 'Asparagus',
  category: 'VEGETABLES',
  priceInCents: 456,
  inStock: 3,
};

export const testProductsQueryResult1: ProductsQueryResult = {
  nodes: [testProduct1, testProduct2],
  pageInfo: {
    page: 1,
    numberOfPages: 5,
  },
};
