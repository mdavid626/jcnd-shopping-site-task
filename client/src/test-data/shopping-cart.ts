import { ShoppingCartItem } from '../types/shopping-cart';
import { testProduct1, testProduct2 } from './products';

export const testShoppingCartItem1: ShoppingCartItem = {
  product: testProduct1,
  amount: 3,
};

export const testShoppingCartItem2: ShoppingCartItem = {
  product: testProduct2,
  amount: 36,
};

export const testShoppingCart1: ShoppingCartItem[] = [
  testShoppingCartItem1,
  testShoppingCartItem2,
];
