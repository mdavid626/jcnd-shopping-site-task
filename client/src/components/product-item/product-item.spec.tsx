import { cleanup, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import {
  useAddToShoppingCart,
  useAvailableInStock,
} from '../../hooks/shopping-car-thooks/shopping-cart-hooks';
import { testProduct1 } from '../../test-data/products';
import ProductItem from './product-item';

jest.mock('../../hooks/shopping-car-thooks/shopping-cart-hooks');

describe('product-item', () => {
  beforeEach(() => {
    (useAddToShoppingCart as jest.Mock).mockReturnValue(jest.fn());
    (useAvailableInStock as jest.Mock).mockReturnValue(12);
  });
  afterEach(cleanup);
  afterEach(jest.resetAllMocks);

  it('should render', () => {
    const { asFragment } = render(<ProductItem product={testProduct1} />);
    expect(asFragment()).toMatchSnapshot();
    expect(useAddToShoppingCart).toHaveBeenCalledWith();
    expect(useAvailableInStock).toHaveBeenCalledWith(testProduct1);
  });

  it('should not render buy button when not available in stock', () => {
    (useAvailableInStock as jest.Mock).mockReturnValue(0);
    render(<ProductItem product={testProduct1} />);
    expect(screen.queryByText('Buy')).toBe(null);
  });

  it('should add to shopping cart when buy button clicked', async () => {
    const addToShoppingCart = jest.fn();
    (useAddToShoppingCart as jest.Mock).mockReturnValue(addToShoppingCart);
    render(<ProductItem product={testProduct1} />);
    await userEvent.click(screen.getByText('Buy'));
    expect(addToShoppingCart).toHaveBeenCalledWith(testProduct1);
  });
});
