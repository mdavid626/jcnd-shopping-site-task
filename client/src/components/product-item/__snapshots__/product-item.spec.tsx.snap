// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`product-item should render 1`] = `
<DocumentFragment>
  <div
    class="ProductItem"
    data-testid="ProductItem-63ea6485732ff5cf8b36442c"
  >
    <div
      class="ProductItem-name"
    >
      Artichoke
    </div>
    <div
      class="ProductItem-priceAndStock"
    >
      46,36 €
      <div
        class="ProductItem-stock"
      >
        In stock: 12
      </div>
    </div>
    <div
      class="ProductItem-buy"
    >
      Buy
    </div>
  </div>
</DocumentFragment>
`;
